modules/core/include/opencv2/core/mat.hpp

```diff
@@ -2281,6 +2323,12 @@ class CV_EXPORTS UMat
     UMat(const UMat& m, const std::vector<Range>& ranges);
     //! builds matrix from std::vector with or without copying the data
     template<typename _Tp> explicit UMat(const std::vector<_Tp>& vec, bool copyData=false);
+
+#ifdef CV_CXX_STD_ARRAY
+    //! builds matrix from std::array with or without copying the data
+    template<typename _Tp, std::size_t _N> explicit UMat(const std::array<_Tp, _N>& vec, bool copyData=false);
+#endif
```

Comment:
Please remove this overload.
UMat usage is highly integrated with OpenCL, which usually requires large enough amount of data (there is significant overhead for small arrays). In general, we should not allow users/developers to write a slow code.